pkg/api/registered/registered.go

```diff
@@ -25,71 +26,97 @@ import (
 	"k8s.io/kubernetes/pkg/api/unversioned"
 )
 
-// List of registered API versions.
-// The list is in the order of most preferred to the least.
-var RegisteredGroupVersions []unversioned.GroupVersion
+var (
+	// registeredVersions represents all enabled API versions. Please call
+	// RegisterVersions() to register the versions.
+	registeredVersions = map[unversioned.GroupVersion]struct{}{}
 
-func init() {
-	validGroupVersions := map[unversioned.GroupVersion]bool{
-		unversioned.GroupVersion{Group: "", Version: "v1"}:                      true,
-		unversioned.GroupVersion{Group: "extensions", Version: "v1beta1"}:       true,
-		unversioned.GroupVersion{Group: "componentconfig", Version: "v1alpha1"}: true,
-		unversioned.GroupVersion{Group: "metrics", Version: "v1alpha1"}:         true,
-	}
-
-	// The default list of supported api versions, in order of most preferred to the least.
-	supportedVersions := []unversioned.GroupVersion{
-		{Group: "", Version: "v1"},
-		{Group: "extensions", Version: "v1beta1"},
-		{Group: "componentconfig", Version: "v1alpha1"},
-	}
+	// envRequestedVersions represents the versions requested via the KUBE_API_VERSIONS
+	// environment variable. The install package of each group checks this list before
+	// add their versions to the latest package and Scheme.
+	envRequestedVersions = map[unversioned.GroupVersion]struct{}{}
```

Comment:
So in keeping with `registered` and `enabled`, this would not affect the `registered` list, but it would prevent versions from becoming `enabled`, right?